trino-aws-proxy-spark3/src/main/java/io/trino/aws/proxy/spark3/TrinoAwsProxyS3ClientSigil.java

```diff
@@ -0,0 +1,19 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.aws.proxy.spark3;
+
+public final class TrinoAwsProxyS3ClientSigil
```

Comment:
If i am understanding this correctly, this class is used just as a pointer to the where the complied files are. Could we not use `[TrinoAwsProxyS3ClientFactory` instead?

Reply:
tbh - I can't remember why I created `TrinoAwsProxyS3ClientSigil`. I'll look into it.

Reply:
@mosiac1 I remember why I did this. It's mostly to get out of dependency hell. The `trino-aws-proxy-spark3` module uses different versions of everything from the rest of the project. This causes a lot of errors/warnings when that module is used as a dependency in the main `trino-aws-proxy` module for testing. A simple solution is to make  `trino-aws-proxy-spark3`'s dependencies `provided`. However, this means that `TrinoAwsProxyS3ClientFactory` can no longer be used because it references a ton of dependencies that are, now, no longer included. So, I added `TrinoAwsProxyS3ClientSigil` which has no dependencies (i.e. no arguments, fields, etc.).

Reply:
Ah I see, that does sound like hell. 
Im wondering if it would help to move the `src/test` classes from `io.trino.aws.proxy.server.testing` (all the utility classes) into `src/main` so that `trino-aws-proxy-spark3` can depend on `trino-aws-proxy-server` for testing rather than the other way around

Reply:
Actually, I can probably move the Sigil to testing and get the directory that way. There's another trick as well - having Maven write out the paths. But, tbh, I thought the Sigil might be useful to others and doesn't add a lot of weight.